File: w1_p5/18.c
#include <stdio.h>
#include <math.h>

int main() {
	int student, num1;
	int i;
	double money[10], average, sum, result, temp;
	
	while(scanf("%d", &student) ==1){
		num1 =student;
		temp = result = average = sum =0;
		if(num1 == 0)break;
		temp = result= sum = average =0.0;
		for(i =0; i< num1; i++){
			scanf("%lf" ,&money[i]);
			 sum = sum + money[i];
		}
		average = sum/num1;
		for(i =0; i <num1; i++){
			if(average > money[i]){
				temp = average- money[i];
				result = temp+ result;
			}
		}
		result = floor(result *100 )/ 100;
		printf("$%.2f\n",result);
	}
}

File: w1_p5/18.java
import java.io.*;
import java.util.Scanner;
class Main {
	public static void main(String[] args) throws Exception {
		Scanner input = new Scanner(System.in);
		
		while(true){
			int n = input.nextInt();
			if (n==0)
				break;
			
			float sum=0;
			float[] a = new float[n];
			for (int i=0; i<n; i++){
				a[i]=input.nextFloat();
				sum+=a[i];
			}
			float avg=sum/n;
			float result=0.00f;
			for (int j=0; j<n; j++){
				float f =(float)(Math.floor((a[j]-avg)*100)/100.0);
				if (f>0) {
					result+=f;
				}
			}
			System.out.printf("$%.2f\n",result);
		}
		input.close();
	}
}